lms/djangoapps/grades/tasks.py

```diff
@@ -24,9 +28,27 @@
 
 @task(default_retry_delay=30, routing_key=settings.RECALCULATE_GRADES_ROUTING_KEY)
 def recalculate_subsection_grade(
+        # pylint: disable=unused-argument
         user_id, course_id, usage_id, only_if_higher, weighted_earned, weighted_possible, **kwargs
 ):
     """
+    Shim to allow us to modify this task's signature without blowing up production on deployment.
+    """
+    recalculate_subsection_grade_v2.apply(
+        kwargs=dict(
+            user_id=user_id,
+            course_id=course_id,
+            usage_id=usage_id,
+            only_if_higher=only_if_higher,
+            expected_modified_time=kwargs.get('expected_modified_time', 0),  # Use the unix epoch as a default
+            score_deleted=kwargs['score_deleted'],
+        )
+    )
+
+
+@task(default_retry_delay=30, routing_key=settings.RECALCULATE_GRADES_ROUTING_KEY)
+def recalculate_subsection_grade_v2(**kwargs):
```

Comment:
Remember that just having `kwargs` in the signature doesn't allow the task to be backward and forward compatible.  Task workers will expect certain arguments to be there for correctness of the feature.  Similarly, enqueuers will expect that their task is being handled by the matching version of the code.

So versioning these tasks are the most full-proof way of handling multi-version deployments.  This assumes that old workers who cannot handle new-versioned tasks will ignore those tasks and leave them for new workers.